sys-variables: update `tidb_schema_cache_size` default value (#19007)
pingcap/docs: system-variables.md

> **Warning:**
>
> The feature controlled by this variable is not GA yet in the current TiDB version. Evaluate it carefully at your own risk.

- Scope: GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Integer
- Default value: `536870912` (512 MiB)
- Range: `0` or `[67108864, 9223372036854775807]`
- Before TiDB v8.4.0, the default value of this variable is `0`.

> - Starting from TiDB v8.4.0, the default value is `536870912`.

Reviewer: ```suggestion
- Starting from TiDB v8.4.0, the default value is `536870912`. The old values will still be used when upgrading from a lower version to v8.4.0 or above.
```

> - Starting from TiDB v8.4.0, the default value is `536870912`. The old values will still be used when upgrading from a lower version to v8.4.0 or above.

Reviewer 2: ```suggestion
- Starting from TiDB v8.4.0, the default value is `536870912`. The old values will still be used when upgrading from a lower version to v8.4.0 or later.
```

> - Starting from TiDB v8.4.0, the default value is `536870912`. When you upgrade from an earlier version to v8.4.0 or later, the old values will still be used.

Reviewer 3: ```suggestion
- Starting from TiDB v8.4.0, the default value is `536870912`. When you upgrade from an earlier version to v8.4.0 or later, the old value set in the earlier version is used.
```

> - This variable controls the size of the schema cache in TiDB. The unit is byte. If the value is `0`, it means the cache limit feature is disabled. To enable this feature, you need to set a value within the range `[67108864, 9223372036854775807]`. TiDB will use this value as the maximum available memory limit and apply the Least Recently Used (LRU) algorithm to cache the required tables, effectively reducing the memory used by schema information.

Reviewer 3: ```suggestion
- This variable controls the size of the schema cache in TiDB. The unit is byte. Setting this variable to `0` means the cache limit feature is disabled. To enable this feature, you need to set a value within the range `[67108864, 9223372036854775807]`. TiDB will use this value as the maximum available memory limit and apply the Least Recently Used (LRU) algorithm to cache the required tables, effectively reducing the memory used by schema information.
```
